### PR TITLE
testing loggers

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Trying to work on #7899

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @carmocca @akihironitta @awaelchli @edward-io @ananthsub @rohitgr7 @kamil-kaczmarek @Raalsky @Blaizzy

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGTM !

File: tests/helpers/boring_model.py [hunk 1]
```diff
@@ -87,22 +87,26 @@ def training_step(...):
...
         out = torch.nn.functional.mse_loss(x, torch.ones_like(x))
         return out
 
     def training_step(self, batch, batch_idx):
-        output = self(batch)
-        loss = self.loss(batch, output)
+        pred = self(batch)
+        loss = self.loss(batch, pred)
+        # self.log("train_loss", loss)
+        # self.log("train_acc", self.train_mse(pred, torch.ones_like(pred)))
```
Comment (ethanwharris) [inline tests/helpers/boring_model.py:109]:
```suggestion
```

File: tests/helpers/boring_model.py [hunk 2]
```diff
@@ -112,16 +116,19 @@ def training_epoch_end(self, outputs) -> None:
...
 
     def validation_epoch_end(self, outputs) -> None:
         torch.stack([x["x"] for x in outputs]).mean()
 
     def test_step(self, batch, batch_idx):
-        output = self(batch)
-        loss = self.loss(batch, output)
+        y_pred = self(batch)
+        loss = self.loss(batch, y_pred)
+        # self.log("test_loss", loss)
```
Comment (ethanwharris) [inline tests/helpers/boring_model.py:131]:
```suggestion
```

**Review by ethanwharris — COMMENTED** [review]
Looks like some debug code that could be removed

File: tests/helpers/boring_model.py [hunk 3]
```diff
@@ -87,22 +87,26 @@ def training_step(...):
...
     def step(self, x):
         x = self(x)
         out = torch.nn.functional.mse_loss(x, torch.ones_like(x))
         return out
 
     def training_step(self, batch, batch_idx):
-        output = self(batch)
-        loss = self.loss(batch, output)
+        pred = self(batch)
+        loss = self.loss(batch, pred)
```
Comment (justusschock) [inline tests/helpers/boring_model.py:107]: Can you revert these changes as the renaming shouldn't be necessary and only increases diff?

File: tests/helpers/boring_model.py [hunk 4]
```diff
@@ -112,16 +116,19 @@ def training_epoch_end(self, outputs) -> None:
         torch.stack([x["loss"] for x in outputs]).mean()
 
     def validation_step(self, batch, batch_idx):
-        output = self(batch)
-        loss = self.loss(batch, output)
+        pred = self(batch)
+        loss = self.loss(batch, pred)
+        # self.log("val_loss", loss)
```
Comment (justusschock) [inline tests/helpers/boring_model.py:121]: also revert this?

File: tests/helpers/boring_model.py [hunk 5]
```diff
@@ -112,16 +116,19 @@ def training_epoch_end(self, outputs) -> None:
...
         return {"x": loss}
 
     def validation_epoch_end(self, outputs) -> None:
         torch.stack([x["x"] for x in outputs]).mean()
 
     def test_step(self, batch, batch_idx):
-        output = self(batch)
-        loss = self.loss(batch, output)
+        y_pred = self(batch)
+        loss = self.loss(batch, y_pred)
```
Comment (justusschock) [inline tests/helpers/boring_model.py:130]: also revert? :)

**Review by justusschock — APPROVED** [review]

File: tests/helpers/boring_model.py (hunk 3 above)
Comment (Borda) [inline tests/helpers/boring_model.py:107]:
```suggestion
        output = self(batch)
        loss = self.loss(batch, output)
```

File: tests/helpers/boring_model.py [hunk 6]
```diff
@@ -112,16 +116,19 @@ def training_epoch_end(self, outputs) -> None:
         torch.stack([x["loss"] for x in outputs]).mean()
 
     def validation_step(self, batch, batch_idx):
-        output = self(batch)
-        loss = self.loss(batch, output)
+        pred = self(batch)
+        loss = self.loss(batch, pred)
```
Comment (Borda) [inline tests/helpers/boring_model.py:120]:
```suggestion
        output = self(batch)
        loss = self.loss(batch, output)
```

File: tests/helpers/boring_model.py (hunk 5 above)
Comment (Borda) [inline tests/helpers/boring_model.py:130]:
```suggestion
        output = self(batch)
        loss = self.loss(batch, output)
```

File: tests/loggers/test_all.py [hunk 7]
```diff
@@ -232,18 +220,7 @@ def name(self):
...
-@pytest.mark.parametrize(
-    "logger_class",
-    [
-        CometLogger,
-        CSVLogger,
-        MLFlowLogger,
-        TensorBoardLogger,
-        TestTubeLogger,
-        # The WandbLogger gets tested for pickling in its own test.
-        # The NeptuneLogger gets tested for pickling in its own test.
```
Comment (awaelchli) [inline tests/loggers/test_all.py:244]: can we preserve this comment? see suggestion below

File: tests/loggers/test_all.py [hunk 8]
```diff
@@ -232,18 +220,7 @@ def name(self):
...
-    ],
-)
+@pytest.mark.parametrize("logger_class", ALL_LOGGER_CLASSES_WO_NEPTUNE)
```
Comment (awaelchli) [inline tests/loggers/test_all.py:223]:
```suggestion
@pytest.mark.parametrize("logger_class", ALL_LOGGER_CLASSES_WO_NEPTUNE)  # WandbLogger and NeptuneLogger get tested separately
```

**Review by awaelchli — APPROVED** [review]

**Review by ethanwharris — APPROVED** [review]
